test/tax-lot/tax-lot.e2e-spec.ts

```diff
@@ -0,0 +1,92 @@
+import * as request from "supertest";
+import { INestApplication } from "@nestjs/common";
+import { Test } from "@nestjs/testing";
+import { TaxLotModule } from "src/tax-lot/tax-lot.module";
+import { TaxLotRepository } from "src/tax-lot/tax-lot.repository";
+import { getTaxLotGeoJsonByBblQueryResponseSchema } from "src/gen";
+import { TaxLotRepositoryMock } from "./tax-lot.repository.mock";
+import {
+  DataRetrievalException,
+  InvalidRequestParameterException,
+} from "src/exception";
+import { HttpName } from "src/filter";
+
+describe("TaxLots", () => {
+  let app: INestApplication;
+
+  const taxLotRepository = new TaxLotRepositoryMock();
+
+  beforeAll(async () => {
+    const moduleRef = await Test.createTestingModule({
+      imports: [TaxLotModule],
+    })
+      .overrideProvider(TaxLotRepository)
+      .useValue(taxLotRepository)
+      .compile();
+    app = moduleRef.createNestApplication();
+    await app.init();
+  });
+
+  describe("getTaxLotGeoJsonByBbl", () => {
+    it("should 200 and return documented schema when finding by valid bbl", async () => {
+      const mock = taxLotRepository.findByBblSpatialMocks[0];
+
+      const response = await request(app.getHttpServer())
+        .get(`/tax-lots/${mock.bbl}/geojson`)
+        .expect(200);
+      expect(
+        getTaxLotGeoJsonByBblQueryResponseSchema.safeParse(response.body)
+          .success,
+      ).toBe(true);
+    });
+
+    it("should 400 when finding by too short bbl", async () => {
+      const shortBbl = "012345678";
+      const response = await request(app.getHttpServer())
+        .get(`/tax-lots/${shortBbl}/geojson`)
+        .expect(400);
+      expect(response.body.message).toBe(
```

Comment:
On error paths, assert the error and/or message match expectations. The expected error structures are written in the `filter` folder